### PR TITLE
Ingredients

### PR DESCRIPTION
**Pull request by akramerin**
- added stimulus controller for the nested forms
- changed recipe create method
- added forms for recipe create
- implemented nested forms for ingredient create
- added recipes show page

### CODE REVIEW

File: app/views/recipes/show.html.erb [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+        <p><%= @recipe.serving %></p>
+        <h5>Difficulty</h5>
+        <%= @recipe.difficulty %> </p>
+        <h5>Utensils:</h5>
+        <p><%= @recipe.utensil %></p>
+      </div>
+        <div class="Ingredients">
+          <h5>Ingredients</h5>
+            <ul>
+              <% @recipe.ingredients.each do |ingredient| %>
```
Comment (syboulette) [inline app/views/recipes/show.html.erb:24]: can we use recipe_ingredient? @akramerin so we stay consisitant :) as they are different! or call them |ri| if you would prefer a shorter version :)

**Review by syboulette — CHANGES_REQUESTED** [review]

File: app/views/recipes/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <%= f.input :prep_time, placeholder: "in minutes" %>
+  <%= f.input :cooking_time, label: "Cook Time", placeholder: "in minutes"  %>
+  <%= f.input :total_time, placeholder: "in minutes"   %>
+  <%= f.input :difficulty, collection: %w[1 2 3 4 5], label: "Difficulty level: 1 = easy, 5 = challenging" %>
+  <%= f.input :dietary_information_recipes, collection: %w[Vegetarian Vegan Pescetarian Gluten-Free Dairy-Free Low-Carb Healthy], label: "Dietary information", as: :check_boxes %>
+  <%= f.input :serving, label: "Servings" %>
+  <%= f.input :photo, as: :file, label: "Show us what it looks like:" %>
+  <%= f.input :utensil, label: "Utensils", placeholder: "A salad bowl" %>
+  <%= f.input :instruction, as: :text, placeholder: "Describe each step of the recipe." %>
+  <%#= f.collection_check_boxes :ingredient_ids, Ingredient.all, :id, :name, include_hidden: false, item_label_class: "ps-1"  %>
```
[the remark this answers is not in the document]
Comment (akramerin) [reply app/views/recipes/_form.html.erb:12]: haha, true, sweet memories of old tries. i am deleting it.

File: app/views/recipes/_ingredient_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="nested-form-wrapper" data-new-record="<%= f.object.new_record? %>">
+  <%= f.input :ingredient_id, collection: Ingredient.all, label_method: :name, label_value: :id, include_hidden: false %>
+  <%= f.input :quantity %>
```
[the remark this answers is not in the document]
Comment (akramerin) [reply app/views/recipes/_ingredient_form.html.erb:3]: added it!

File: app/views/recipes/new.html.erb [hunk 4]
```diff
@@ -1,30 +1,18 @@
...
-        <h6 <em><%= recipe_ingredient.unit %></em></h6>
-        <div class="icon">
-          <%= link_to image_tag("edit.svg"), "/recipes/#{recipe_ingredient.recipe_id}/recipe_ingredients/#{recipe_ingredient.id}/edit" %>
-          <%= link_to image_tag("bin.svg"), recipe_ingredient_path(recipe_ingredient), method: :delete %>
-        </div>
-      </div>
+    <ul>
+      <h6><%= recipe_ingredient.ingredient.name %></h6>
+      <h6><%= recipe_ingredient.quantity %></h6>
+    </li>
```
[the remark this answers is not in the document]
Comment (akramerin) [reply app/views/recipes/new.html.erb:11]: actually this entire part was unnecessary thanks to the magic of nested forms - i deleted it.

File: app/views/recipes/show.html.erb [hunk 5]
```diff
@@ -1,22 +1,34 @@
...
+        <h5>Total Time</h5>
+        <p><%= @recipe.total_time %> minutes</p>
+        <i class="fa-duotone fa-cake-slice"></i> <h5>Servings</h5>
+        <p><%= @recipe.serving %></p>
+        <h5>Difficulty</h5>
+        <%= @recipe.difficulty %> </p>
+        <h5>Utensils:</h5>
+        <p><%= @recipe.utensil %></p>
+      </div>
+        <div class="Ingredients">
```
[the remark this answers is not in the document]
Comment (akramerin) [reply app/views/recipes/show.html.erb:21]: nope, fixed it

File: app/views/recipes/show.html.erb (hunk 1 above)
Comment (akramerin) [reply app/views/recipes/show.html.erb:24]: went with "recipe_ingredient" :)

**Comment by akramerin** [comment]
I fixed the suggested changes @syboulette - and also some errors in the view and added the edit a recipe view.

**Review by syboulette — APPROVED** [review]

**Pull request merged**